packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -712,6 +715,48 @@ export const useActionGroup: EnsembleActionHook<ExecuteActionGroupAction> = (
   return { callback };
 };
 
+export const useDispatchEvent: EnsembleActionHook<DispatchEventAction> = (
+  action,
+) => {
+  const eventName = keys(action)[0];
+  const [isComplete, setIsComplete] = useState<boolean>();
+  const [context, setContext] = useState<unknown>();
+  const eventData = (action ? action[eventName] : {}) as {
+    [key: string]: unknown;
+  };
+  const eventScopre = useCustomEventScope();
+  const evaluatedInputs = useEvaluate(eventData, { context });
+
+  const events = get(eventScopre, eventName) as {
+    [key: string]: unknown;
+  };
+
+  const eventActions = Object.keys(events || {}).map((customAction) => ({
+    [customAction]: events[customAction],
+  }));
+
+  // Use a separate hook call for each event action
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const ensembleActions = eventActions.map((event) => useEnsembleAction(event));
+
+  const callback = useCallback((args: unknown): void => {
+    setContext(args);
+    setIsComplete(false);
+  }, []);
+
+  useEffect(() => {
+    if (isComplete !== false) {
+      return;
+    }
+
+    ensembleActions.forEach((act) => act?.callback(evaluatedInputs));
```

Comment:
Context also needs to be passed in?

Reply:
@evshi We are already passing the context as evauluatedInputs, which will evaluated from the given arguments while calling the actions

Reply:
Right but sometimes the arguments need to be passed through to the following action, e.g. `files` or `response`